scarb/src/core/publishing/manifest_normalization.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::BTreeMap;
+
+use anyhow::{bail, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use indoc::formatdoc;
+
+use crate::core::{
+    DepKind, DependencyVersionReq, DetailedTomlDependency, ManifestDependency, MaybeWorkspace,
+    Package, PackageName, TomlDependency, TomlManifest, TomlPackage, TomlWorkspaceDependency,
+    TomlWorkspaceField,
+};
+
+pub fn prepare_manifest_for_publish(pkg: &Package) -> Result<TomlManifest> {
+    let package = Some(generate_package(pkg));
+
+    let dependencies = Some(generate_dependencies(
+        // NOTE: We deliberately do not ask for `full_dependencies` here, because
+        // we do not want to emit requirements for built-in packages like `core`.
+        &pkg.manifest.summary.dependencies,
+        DepKind::Normal,
+    )?);
+
+    let tool = pkg.manifest.metadata.tool_metadata.clone().map(|m| {
+        m.into_iter()
+            .map(|(k, v)| (k, MaybeWorkspace::Defined(v)))
+            .collect()
+    });
+
+    Ok(TomlManifest {
+        package,
+        workspace: None,
+        dependencies,
+        lib: None,
+        // TODO(mkaput): Allow packaging Cairo plugins.
+        cairo_plugin: None,
+        test: None,
+        target: None,
```

Comment:
Shouldn't we include package targets? 🤔

Reply:
They are not needed currently. We are only interested in dependencies, which are inconstructible currently anyway, so I didn't bother with them here.